### DISCUSSION TITLE
Generate JavaScript configuration file for accessing colors/sizes (screen, font, etc.) from code

### DISCUSSION THREAD

**Discussion by szabototo89** [tailwindlabs/tailwindcss#2373] [Ideas]
I'd like to have more interaction with TailwindCSS and I'd be great if we could access to actual configuration from a generated file. For example, I'm working a React application where I need to use media queries via window.matchMedia from TypeScript and I use TailwindCSS at the same time. In order to have consistent data (aka a single source of truth), TailwindCSS could generate a configuration file storing all of the current values (default and customized) for each value.

**Comment by adamwathan** [comment]
Check out this section of the docs:

https://tailwindcss.com/docs/configuration#referencing-in-java-script

Works really well in combination with babel-preval to minimize impact on bundle size 👍🏻
